openstack/clustering/v1/clusters/results.go

```diff
@@ -0,0 +1,127 @@
+package clusters
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+	"time"
+
+	"reflect"
+
+	"github.com/gophercloud/gophercloud"
+)
+
+// commonResult is the response of a base result.
+type commonResult struct {
+	gophercloud.Result
+}
+
+// CreateResult is the response of a Create operations.
+type CreateResult struct {
+	commonResult
+}
+
+type Cluster struct {
+	Config          map[string]interface{} `json:"config"`
+	CreatedAt       time.Time              `json:"-"`
+	Data            map[string]interface{} `json:"data"`
+	Dependents      map[string]interface{} `json:"dependents"`
+	DesiredCapacity int                    `json:"desired_capacity"`
+	Domain          string                 `json:"domain"`
+	ID              string                 `json:"id"`
+	InitAt          time.Time              `json:"-"`
+	MaxSize         int                    `json:"max_size"`
+	Metadata        map[string]interface{} `json:"metadata"`
```

Comment:
The Senlin code shows this as a JSON field:

https://github.com/openstack/senlin/blob/e5bc16b3fea2f10dead365f983c5a937f23d72b0/senlin/objects/cluster.py#L45

Can you add some metadata to the unit tests to ensure it's parsed correctly?

Reply:
Done